frontend_csv/src/parser/ast_msg.rs

```diff
@@ -32,38 +33,38 @@ pub struct AstMsg {
     pub msg_val: String,
 }
 
-pub struct InitMsg {
+pub struct Msg {
     pub raw: Vec<(AstMsgLineType, Vec<String>)>,
     pub msg_type: AstMsgType,
-    pub msg_val: String,
+    pub msg_val: Vec<Vec<String>>,
     pub msg_name: String,
     pub tlvs_name: Vec<String>,
 }
 
 pub trait AstMsgTrait {
     fn get_msg_type(&self) -> AstMsgType;
-    fn get_values(&self) -> String;
+    fn get_values(&self) -> Vec<Vec<String>>;
     fn check_valid(&mut self) -> bool;
 }
 
-impl InitMsg {
-    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> InitMsg {
-        return InitMsg {
-            raw: lines,
-            msg_type: AstMsgType::Init,
-            msg_val: String::new(),
+impl Msg {
+    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> Msg {
+        return Msg {
+            raw: lines.clone(),
+            msg_type: AstMsgType::None,
```

Comment:
```suggestion
            msg_type: None,
```

you can use the rust option type! and in this case it is better, ou can remove `AstMsgType::None` because cause confusion and corner case